tests/components/snmp/test_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""SNMP sensor tests."""
+
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+
+
+@pytest.fixture(autouse=True)
+def hlapi_mock():
+    """Mock out 3rd party API."""
+    mock_data = MagicMock()
+    mock_data.prettyPrint = Mock(return_value="hello")
+    with patch(
+        "homeassistant.components.snmp.sensor.getCmd",
+        return_value=(None, None, None, [[mock_data]]),
+    ):
+        yield
+
+
+async def test_basic_config(hass: HomeAssistant) -> None:
+    """Test basic entity configuration."""
+
+    config = {
+        SENSOR_DOMAIN: {
+            "platform": "snmp",
+            "host": "192.168.1.32",
+            "baseoid": "1.3.6.1.4.1.2021.10.1.3.1",
+        },
+    }
+
+    assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.snmp")
+    assert state.state == "hello"
+    assert state.attributes == {"friendly_name": "SNMP"}
+
+
+async def test_entity_config(hass: HomeAssistant) -> None:
+    """Test entity configuration."""
+
+    config = {
+        SENSOR_DOMAIN: {
+            # SNMP configuration
+            "platform": "snmp",
+            "host": "192.168.1.32",
+            "baseoid": "1.3.6.1.4.1.2021.10.1.3.1",
+            # Entity configuration
+            "attributes": {"extra1": "{{'extra 1'}}"},
```

Comment:
I'm not sure this is wanted?

Reply:
I don't think we should include support for attributes by default. We're trying to avoid using state attributes.

Reply:
Removed in 730238a